File: test_regress/t/t_leak.pl
#!/usr/bin/env perl
if (!$::Driver) { use FindBin; exec("$FindBin::Bin/bootstrap.pl", @ARGV, $0); die; }
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2003 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

if ($Self->{vltmt} && exists $ENV{TRAVIS_DIST} &&
    $ENV{TRAVIS_DIST} eq "trusty")
{
    skip("Multithreaded test does not work under CI w/ Ubuntu Trusty");
}

scenarios(vlt_all => 1);

compile(
    make_top_shell => 0,
    make_main => 0,
    verilator_flags2 => ["--exe $Self->{t_dir}/$Self->{name}.cpp"],
    make_flags => 'CPPFLAGS_ADD=-DVL_NO_LEGACY',
    );

execute(
    check_finished => 1,
    );

ok(1);
1;

File: test_regress/t/t_leak.py
#!/usr/bin/env python3
# DESCRIPTION: Verilator: Verilog Test driver/expect definition
#
# Copyright 2024 by Wilson Snyder. This program is free software; you
# can redistribute it and/or modify it under the terms of either the GNU
# Lesser General Public License Version 3 or the Perl Artistic License
# Version 2.0.
# SPDX-License-Identifier: LGPL-3.0-only OR Artistic-2.0

import vltest_bootstrap

test.scenarios('vlt_all')

if test.vltmt and test.getenv_def('TRAVIS_DIST', "None") == "trusty":
    test.skip("Multithreaded test does not work under CI w/ Ubuntu Trusty")

test.compile(make_top_shell=False,
             make_main=False,
             verilator_flags2=["--exe", test.pli_filename],
             make_flags=['CPPFLAGS_ADD=-DVL_NO_LEGACY'])

test.execute()

test.passes()
